subprojects/packagefiles/libssh2/meson.build

```diff
@@ -0,0 +1,196 @@
+project('libssh2', 'c',
+  version : '1.10.0',
+  license : 'BSD-3',
+  meson_version : '>=0.58.0')
+
+cc = meson.get_compiler('c')
+pkgconfig = import('pkgconfig')
+
+libssh2_sources = [
+    'src/misc.c',
+    'src/scp.c',
+    'src/crypt.c',
+    'src/hostkey.c',
+    'src/version.c',
+    'src/wincng.c',
+    'src/blowfish.c',
+    'src/mac.c',
+    'src/os400qc3.c',
+    'src/transport.c',
+    'src/agent.c',
+    'src/pem.c',
+    'src/kex.c',
+    'src/sftp.c',
+    'src/knownhost.c',
+    'src/publickey.c',
+    'src/channel.c',
+    'src/global.c',
+    'src/packet.c',
+    'src/bcrypt_pbkdf.c',
+    'src/userauth.c',
+    'src/keepalive.c',
+    'src/comp.c',
+    'src/session.c'
+]
+
+libssh2_deps = []
+with_openssl = false
+with_embedtls = false
+with_gcrypt = false
+extra_flags = []
+if get_option('crypto_backend') == 'openssl'
+    with_openssl = true
+    libssh2_sources += ['src/openssl.c']
+
+    openssl_dep = dependency('openssl', version : '>= 1.0.0')
+    libssh2_deps += openssl_dep
+
+    extra_flags += ['-DLIBSSH2_OPENSSL']
+elif get_option('crypto_backend') == 'embedtls'
+    with_embedtls = true
+    libssh2_sources += ['src/mbedtls.c']
+
+    embedtls_dep = dependency('embedtls')
+
+    libssh2_deps += embedtls_dep
+    extra_flags += ['-DLIBSSH2_EMBEDTLS']
+elif get_option('crypto_backend') == 'gcrypt'
+    with_gcrypt = true
+    libssh2_sources += ['src/libgcrypt.c']
+
+    gcrypt_dep = dependency('gcrypt')
+
+    libssh2_deps += gcrypt_dep
+    extra_flags += ['-DLIBSSH2_GCRYPT']
+else
+    error('No crypto backend specified')
+endif
+
+libssh2_have_unistd_h = cc.has_header('unistd.h')
```

Comment:
You could use a loop for this. example: https://github.com/mesonbuild/wrapdb/blob/master/subprojects/packagefiles/libsndfile/meson.build <-- 70

Reply:
Done